src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -197,10 +210,33 @@ impl<S: LocalStateStore> WriteFuture<S> {
         }
     }
 
+    fn paused(
+        duration: Duration,
+        message: Message,
+        stream: BoxedMessageStream,
+        write_state: LogStoreWriteState<S>,
+    ) -> Self {
+        Self::Paused {
+            sleep_future: Some(Box::pin(sleep(duration))),
+            message,
+            stream,
+            write_state,
+        }
+    }
+
     async fn next_event(
         &mut self,
     ) -> StreamExecutorResult<(BoxedMessageStream, LogStoreWriteState<S>, WriteFutureEvent)> {
         match self {
+            WriteFuture::Paused { sleep_future, .. } => {
```

Comment:
Seems no need to make it `Option`? Because we always overwrite it with WriteFuture::Empty after it returns successfully.